src/foam/nanos/column/NestedPropertiesExpression.js

```diff
@@ -81,61 +83,124 @@ foam.CLASS({
         {
           name: 'i',
           class: 'Int'
+        },
+        {
+          name: 'expr',
+          javaType: 'foam.mlang.Expr'
         }
       ],
-      code: function (of, propName, i) {
-        if ( i === propName.length - 1 )
-          return of.getAxiomByName(propName[i]);
+      code: function (of, propName, i, expr) {
         var prop = of.getAxiomByName(propName[i]);
-        return foam.mlang.Expressions.create().DOT(prop, this.returnPropExpr(prop, this.returnDotExprForNestedProperty(prop.of, propName, ++i)));
+        if ( ! prop ) return null;
+
+        var propExpr = this.buildPropertyExpr(prop, expr);
+
+        if ( i === propName.length - 1 )
+          return propExpr;
+
+        return this.returnDotExprForNestedProperty(prop.of, propName, ++i, propExpr);
       },
       javaCode: `
-        ClassInfo ci = of;
-        PropertyInfo p = (PropertyInfo) ci.getAxiomByName(propName[i]);
-        if ( i == propName.length - 1 ) {
-          return p;
-        }
+        PropertyInfo prop = (PropertyInfo) of.getAxiomByName(propName[i]);
+
+        if ( prop == null ) return null;
+
+        Method m = getFinderMethod(prop);
+        expr = buildPropertyExpr(prop, expr);
+
+        if ( i == propName.length - 1 )
+          return expr;
         
-        StringBuilder sb = new StringBuilder("find");
         try {
-          Class cls;
-          if ( p instanceof foam.core.AbstractFObjectPropertyInfo ) {
-            cls = p.getValueClass();
-          } else {
-            sb.setLength(4);
-            Method m = ci.getObjClass().getMethod(StringUtil.capitalize(p.getName()), foam.core.X.class);
-            cls = m.getReturnType();
-          }
-          ci = (ClassInfo) cls.getMethod("getOwnClassInfo").invoke(null);
-        } catch (Exception e) {
-          Logger logger = (Logger) getX().get("logger");
-          logger.error(e);
+          ClassInfo ci = getPropertyClassInfo(prop);
+          return returnDotExprForNestedProperty(ci, propName, ++i, expr);
+        } catch ( Throwable t ) {
           return null;
         }
+      `
+    },
+    {
+      name: 'getPropertyClassInfo',
+      javaType: 'ClassInfo',
+      javaThrows: [
+        'NoSuchMethodException',
+        'IllegalAccessException',
+        'InvocationTargetException'
```

Comment:
should be `javaThrows: [ 'java.lang.Exception' ]` because line 141 can return null.

Reply:
fixed